File: js/leetCodeEasy/longestCommonPrefix.js
// 14. Longest Common Prefix
// https://leetcode.com/problems/longest-common-prefix/

/*
Write a function to find the longest common prefix string amongst an array of strings.
If there is no common prefix, return an empty string "".

Example 1:
Input: ["flower","flow","flight"]
Output: "fl"

Example 2:
Input: ["dog","racecar","car"]
Output: ""

Explanation: There is no common prefix among the input strings.
*/

export const longestCommonPrefix = (strs) => {
    if(strs.length === 0) {
        return "";
    }

    if(strs.length === 1) {
        return strs[0];
    }

    let result = strs[0];

    for(let i = 1; i < strs.length; i++) {
        for(let j = 0; j < result.length; j++) {
            if(strs[i].charAt(j) !== result.charAt(j)) {
                result = result.slice(0, j);
                break;
            }
        }
    }

    return result;
};

File: go/leetcode_easy/longest_common_prefix/longest_common_prefix.go
package longest_common_prefix

// 14. Longest Common Prefix
// https://leetcode.com/problems/longest-common-prefix/

func LongestCommonPrefix(strs []string) string {
	if len(strs) == 0 {
		return ""
	}

	if len(strs) == 1 {
		return strs[0]
	}

	result := strs[0]

	for i := 1; i < len(strs); i++ {
		for j := 0; j < len(result); j++ {
			if len(strs[i]) == j || strs[i][j] != result[j] {
				result = result[0:j]
				break
			}
		}
	}

	return result
}
